src/renderers/dom/client/syntheticEvents/SyntheticEvent.js

```diff
@@ -195,3 +198,44 @@ SyntheticEvent.augmentClass = function(Class, Interface) {
 PooledClass.addPoolingTo(SyntheticEvent, PooledClass.fourArgumentPooler);
 
 module.exports = SyntheticEvent;
+
+
+// Utility functions
+
+/**
+  * Helper to nullify syntheticEvent instance properties when destructing
+  *
+  * @param {object} SyntheticEvent
+  * @param {String} propName
+  */
+function getPooledWarningPropertyDefinition(propName, getVal) {
+  var setTo = typeof getVal === 'function' ? 'a no-op function' : 'set to null';
+  return {
+    configurable: true,
+    set: function(val) {
+      var warningCondition = false;
+      warning(
+        warningCondition,
+        'This synthetic event is reused for performance reasons. If you\'re ' +
+        'seeing this, you\'re setting property `%s` on a ' +
+        'released/nullified synthetic event. This is effectively a no-op. See ' +
+        'https://fb.me/react-event-pooling for more information.',
+        propName
+      );
+      return val;
+    },
+    get: function() {
+      var warningCondition = false;
+      warning(
+        warningCondition,
+        'This synthetic event is reused for performance reasons. If you\'re ' +
+        'seeing this, you\'re accessing property `%s` on a ' +
+        'released/nullified synthetic event. This is %s. See ' +
+        'https://fb.me/react-event-pooling for more information.',
```

Comment:
That's reasonable. Updating now :-)

On Sat, Jan 30, 2016 at 1:00 PM Dan Abramov <notifications@github.com>
wrote:

> In src/renderers/dom/client/syntheticEvents/SyntheticEvent.js
> <https://github.com/facebook/react/pull/5940#discussion_r51350027>:
>
> > +        'This synthetic event is reused for performance reasons. If you\'re ' +
> > +        'seeing this, you\'re setting property `%s` on a ' +
> > +        'released/nullified synthetic event. This is effectively a no-op. See ' +
> > +        'https://fb.me/react-event-pooling for more information.',
> > +        propName
> > +      );
> > +      return val;
> > +    },
> > +    get: function() {
> > +      var warningCondition = false;
> > +      warning(
> > +        warningCondition,
> > +        'This synthetic event is reused for performance reasons. If you\'re ' +
> > +        'seeing this, you\'re accessing property `%s` on a ' +
> > +        'released/nullified synthetic event. This is %s. See ' +
> > +        'https://fb.me/react-event-pooling for more information.',
>
> Ah, good point. Maybe something like "accessing a method" is neutral
> enough?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/react/pull/5940/files#r51350027>.
>
-- 
- Kent C. Dodds <https://twitter.com/kentcdodds>